### DISCUSSION TITLE
Input elements with buttons, tabs and sql highlight

### DISCUSSION THREAD

**Discussion by steph258** [elsa-workflows/elsa-core#3125] [Q&A]
i am trying a few things and have some questions:

1. is it possible to get buttons for input elements like system functions for choosing files or paths? (for custom activities)
2. is there a way to add custom tabs to the dashboard or to integrate it into other projects?
3. i updated to 2.8+ to get sql syntax highlighting and it works but after the execution of the activity it says the activity faulted with an exception:  the given key sql was not present in the dictionary.

**Comment by mohdali** [comment]
For point no 3, if you are adding SQL synatx to your custom activity, you need to do the following:

- Add reference to the new package `Elsa.Scripting.Sql`
- Add the SQL handler by including the below line in your extension methods or elsa configuration:

`elsa.AddSqlScripting();`
